components/org.wso2.carbon.identity.application.authenticator.fido2/src/main/java/org/wso2/carbon/identity/application/authenticator/fido2/core/WebAuthnService.java

```diff
@@ -163,8 +165,14 @@
         User user = User.getUserFromUserName(CarbonContext.getThreadLocalCarbonContext().getUsername());
         user.setTenantDomain(CarbonContext.getThreadLocalCarbonContext().getTenantDomain());
 
-        PublicKeyCredentialCreationOptions credentialCreationOptions = relyingParty
-                .startRegistration(buildStartRegistrationOptions(user));
+//        PublicKeyCredentialCreationOptions credentialCreationOptions = relyingParty
+//                .startRegistration(buildStartRegistrationOptions(user));
+
+        PublicKeyCredentialCreationOptions credentialCreationOptions =
+                relyingParty.startRegistration(StartRegistrationOptions.builder().user(UserIdentity.builder()
+                        .name(user.toString()).displayName(user.getUserName()).id(generateRandom()).build())
```

Comment:
So is this the full qualified user name that gets resolved as the user name

Reply:
Yes. This is the user information that is used in the device registration ceremony.